src/iov.c

```diff
@@ -67,3 +67,25 @@ uint64_t ofi_copy_iov_buf(const struct iovec *iov, size_t iov_count, uint64_t io
 	}
 	return done;
 }
+
+void ofi_adjust_iovec(struct iovec *iovec, size_t *iovec_count, size_t offset)
+{
+	uint64_t buf_offset = offset;
+	int new_iov = 0, i;
+	size_t len = 0;
+
+	for (i = 0 ; i < *iovec_count ; i++) {
+		len += iovec[i].iov_len;
+
+		if (offset < len)
+			continue;
+
+		new_iov = i+1;
+		buf_offset = offset - len;
+	}
+
+	iovec[new_iov].iov_base = ((uint8_t *)(iovec[new_iov].iov_base) +
+				   buf_offset);
+	iovec[new_iov].iov_len -= buf_offset;
+	(*iovec_count) -= new_iov;
+}
```

Comment:
This function doesn't look correct., or I'm not understanding what it's attempting to do.  The following code is closer to what I think is needed (with an optimization for the single iovec case - renames to match similar code in this file):
```
void ofi_consume_iov(struct iovec *iov, size_t *iov_count, size_t consumed)
{
    size_t i;

    if (*iov_count == 1)
        goto out;

    for (i = 0; i < *iov_count; i++) {
        if (consumed < iov[i].iov_len)
            break;
        consumed -= iov[i].iov_len;
    }
    memmove(iov, &iov[i], sizeof(*iov) * (*iov_count - i));
    *iov_count -= i;
out:
    iov[0].iov_base += consumed;
    iov[0].iov_len -= consumed;
}
```
I don't see a memmove type operation in the above code that would 'shift' the iovec entries forward.  Either that's missing or I don't understand what the intent of this function is.

Reply:
Yes the memmove should be there. I will just use your function. thanks